src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -546,8 +546,12 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return kernel;
 	}
 
-	private getDisplayNameFromSpecName(kernelid: string): string {
-		let newKernel = this.notebookManager.sessionManager.specs.kernels.find(kernel => kernel.name === kernelid);
+	private getDisplayNameFromSpecName(kernel: nb.IKernel): string {
+		let specs = this.notebookManager.sessionManager.specs;
+		if (!specs || !specs.kernels) {
+			return kernel.name;
+		}
+		let newKernel = this.notebookManager.sessionManager.specs.kernels.find(kernel => kernel.name === kernel.name);
```

Comment:
Just checking if we can use a different alias in condition checking. Both the names are accessed from kernel object.

Reply:
Sorry Raj can you clarify? what's the suggested change?

Reply:
kernel.name === kernel.name.. little confusing from readability.

Reply:
I think this might be a bug, need to check